src/main/java/org/wso2/carbon/esb/connector/NatsConfigConnector.java

```diff
@@ -0,0 +1,98 @@
+        String pedantic = validateParameter((String) messageContext.getProperty(NatsConstants.PEDANTIC));
+        String supportUtf8Subjects = validateParameter((String) messageContext.getProperty(NatsConstants.SUPPORT_UTF8_SUBJECTS));
+        String useOldRequestStyle = validateParameter((String) messageContext.getProperty(NatsConstants.USE_OLD_REQUEST_STYLE));
+        String noRandomize = validateParameter((String) messageContext.getProperty(NatsConstants.NO_RANDOMIZE));
+        String noEcho = validateParameter((String) messageContext.getProperty(NatsConstants.NO_ECHO));
+        String maxPoolSize = validateMaxPoolSize((String) messageContext.getProperty(NatsConstants.MAX_CONNECTION_POOL_SIZE));
+
+        messageContext.setProperty(NatsConstants.USERNAME, username);
+        messageContext.setProperty(NatsConstants.PASSWORD, password);
+        messageContext.setProperty(NatsConstants.TLS_PROTOCOL, tlsProtocol);
+        messageContext.setProperty(NatsConstants.TLS_KEYSTORE_TYPE, tlsKeyStoreType);
+        messageContext.setProperty(NatsConstants.TLS_KEYSTORE_LOCATION, tlsKeyStoreLocation);
+        messageContext.setProperty(NatsConstants.TLS_KEYSTORE_PASSWORD, tlsKeyStorePassword);
+        messageContext.setProperty(NatsConstants.TLS_TRUSTSTORE_TYPE, tlsTrustStoreType);
+        messageContext.setProperty(NatsConstants.TLS_TRUSTSTORE_LOCATION, tlsTrustStoreLocation);
+        messageContext.setProperty(NatsConstants.TLS_TRUSTSTORE_PASSWORD, tlsTrustStorePassword);
+        messageContext.setProperty(NatsConstants.TLS_KEY_MANAGER_ALGORITHM, tlsKeyManagerAlgorithm);
+        messageContext.setProperty(NatsConstants.TLS_TRUST_MANAGER_ALGORITHM, tlsTrustManagerAlgorithm);
+        messageContext.setProperty(NatsConstants.CONNECTION_NAME, connectionName);
+        messageContext.setProperty(NatsConstants.VERBOSE, verbose);
+        messageContext.setProperty(NatsConstants.PEDANTIC, pedantic);
+        messageContext.setProperty(NatsConstants.SUPPORT_UTF8_SUBJECTS, supportUtf8Subjects);
+        messageContext.setProperty(NatsConstants.USE_OLD_REQUEST_STYLE, useOldRequestStyle);
+        messageContext.setProperty(NatsConstants.NO_RANDOMIZE, noRandomize);
+        messageContext.setProperty(NatsConstants.NO_ECHO, noEcho);
+        messageContext.setProperty(NatsConstants.MAX_CONNECTION_POOL_SIZE, maxPoolSize);
+    }
+
+    /**
+     * Validate null parameter.
+     *
+     * @param parameter the parameter to validate.
+     * @return the updated parameter.
+     */
+    private String validateParameter(String parameter) {
+        return (parameter == null) ? "" : parameter;
+    }
+
+    /**
+     * Validate maximum pool size parameter.
+     *
+     * @param maxPoolSize the maximum pool size to validate.
+     * @return the updated maximum pool size.
+     */
+    private String validateMaxPoolSize(String maxPoolSize) {
+        try {
+            if (!(maxPoolSize != null && (Integer.parseInt(maxPoolSize) > 1))) {
```

Comment:
Isn't this the same as 'maxPoolSize == null || (Integer.parseInt(maxPoolSize) < 1))' ? IMO it will more readable if we can write it this way.